sdk/resourcemanager/Azure.ResourceManager.Core/src/Generated/PreDefinedTagsOperations.cs

```diff
@@ -0,0 +1,163 @@
+        {
+            using var scope = Diagnostics.CreateScope("PreDefinedTagsOperations.DeleteValue");
+            scope.Start();
+            try
+            {
+                return RestClient.DeleteValue(tagName, tagValue, cancellationToken);
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
+        }
+
+        /// <summary> This operation allows adding a value to the list of predefined values for an existing predefined tag name. A tag value can have a maximum of 256 characters. </summary>
+        /// <param name="tagName"> The name of the tag. </param>
+        /// <param name="tagValue"> The value of the tag to create. </param>
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        public virtual async Task<Response<PreDefinedTagValue>> CreateOrUpdateValueAsync(string tagName, string tagValue, CancellationToken cancellationToken = default)
+        {
+            using var scope = Diagnostics.CreateScope("PreDefinedTagsOperations.CreateOrUpdateValue");
+            scope.Start();
+            try
+            {
+                return await RestClient.CreateOrUpdateValueAsync(tagName, tagValue, cancellationToken).ConfigureAwait(false);
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
+        }
+
+        /// <summary> This operation allows adding a value to the list of predefined values for an existing predefined tag name. A tag value can have a maximum of 256 characters. </summary>
+        /// <param name="tagName"> The name of the tag. </param>
+        /// <param name="tagValue"> The value of the tag to create. </param>
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        public virtual Response<PreDefinedTagValue> CreateOrUpdateValue(string tagName, string tagValue, CancellationToken cancellationToken = default)
+        {
+            using var scope = Diagnostics.CreateScope("PreDefinedTagsOperations.CreateOrUpdateValue");
+            scope.Start();
+            try
+            {
+                return RestClient.CreateOrUpdateValue(tagName, tagValue, cancellationToken);
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
+        }
+
+        /// <summary> This operation allows deleting a name from the list of predefined tag names for the given subscription. The name being deleted must not be in use as a tag name for any resource. All predefined values for the given name must have already been deleted. </summary>
+        /// <param name="tagName"> The name of the tag. </param>
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        public virtual async Task<Response> DeleteAsync(string tagName, CancellationToken cancellationToken = default)
```

Comment:
should have the operation variants for consistency

Reply:
Done